instrumentation/httptrace/config.go

```diff
@@ -39,6 +39,8 @@ type config struct {
 	traceClientIP                bool
 	isStatusError                func(statusCode int) bool
 	inferredProxyServicesEnabled bool
+	allowAllBaggage              bool
+	baggageTagKeys               map[string]struct{}
```

Comment:
I assume you're making baggageTagKeys a `map` to function like a set, in the Golang way? I was surprised at first, would be good to annotate this key with information about its value and why its data structure is what it is